pkg/dispers/dispers/requests.go

```diff
@@ -26,4 +37,201 @@ type InputCI struct {
 // OutputCI contains a bool and the result
 type OutputCI struct {
 	Hashes []Concept `json:"hashes,omitempty"`
+
+/*
+*
+Target Finders' Input & Output
+*
+*/
+
+func union(a, b []string) []string {
+	m := make(map[string]bool)
+
+	for _, item := range a {
+		m[item] = true
+	}
+
+	for _, item := range b {
+		if _, ok := m[item]; !ok {
+			a = append(a, item)
+		}
+	}
+	return a
+}
+
+func intersection(a, b []string) (c []string) {
+	m := make(map[string]bool)
+
+	for _, item := range a {
+		m[item] = true
+	}
+
+	for _, item := range b {
+		if _, ok := m[item]; ok {
+			c = append(c, item)
+		}
+	}
+	return
+}
+
+// Operation allows the possibility to compute targer profile in a
+// recursive way
+type Operation struct {
+	Type   string      `json:"type,omitempty"`
+	Value  string      `json:"value,omitempty"`
+	ValueA interface{} `json:"value_a,omitempty"`
+	ValueB interface{} `json:"value_b,omitempty"`
+}
+
+// Compute return the list of encrypted addresses
+func (o *Operation) Compute(list map[string][]string) ([]string, error) {
+
+	if o.Type == "" {
+		return []string{}, errors.New("Unknown type")
+	}
+
+	// Retrieve list of addresses if single
+	if o.Type == single {
+		val, ok := list[o.Value]
+		if !ok {
+			return []string{}, errors.New("Unknown concept")
+		}
+		return val, nil
+	}
+
+	// Compute operation A and B
+	vA := o.ValueA.(Operation)
+	a, err := vA.Compute(list)
+	if err != nil {
+		return []string{}, err
+	}
+	vB := o.ValueB.(Operation)
+	b, err := vB.Compute(list)
+	if err != nil {
+		return []string{}, err
+	}
+	switch o.Type {
+	case "union":
+		return union(a, b), nil
+	case "intersection":
+		return intersection(a, b), nil
+	default:
+		return []string{}, errors.New("Unknown type")
+	}
+}
+
+// MarshalJSON marhsal an operation of type Single
+func (o *Operation) MarshalJSON() ([]byte, error) {
```

Comment:
Since the Target Profile is made by the Querier, this function is used by the Querier. I needed it here to test UnmarshalJSON() and to check that all this Operation structure is viable.